server/sonar-web/src/main/js/app/components/nav/global/GlobalNavSearchForm.js

```diff
@@ -0,0 +1,350 @@
+    }));
+    return uniqBy([...components, ...recentlyBrowsed], 'key');
+  };
+
+  fetchFavorites = () => {
+    this.setState({ loading: true });
+    getFavorites().then(response => {
+      const results = groupBy(
+        this.mergeWithRecentlyBrowsed(
+          response.favorites.map(component => ({ ...component, isFavorite: true }))
+        ),
+        'qualifier'
+      );
+      const list = this.getPlainComponentsList(results);
+      this.setState({
+        loading: false,
+        more: {},
+        results,
+        selected: list.length > 0 ? list[0].key : null
+      });
+    });
+  };
+
+  search = (query: string) => {
+    this.setState({ loading: true });
+    const recentlyBrowsed = RecentHistory.get().map(component => component.key);
+    getSuggestions(query, recentlyBrowsed).then(response => {
+      const results = {};
+      const more = {};
+      response.results.forEach(group => {
+        results[group.q] = group.items.map(item => ({ ...item, qualifier: group.q }));
+        more[group.q] = group.more;
+      });
+      const list = this.getPlainComponentsList(results);
+      this.setState(state => ({
+        loading: false,
+        more,
+        organizations: { ...state.organizations, ...keyBy(response.organizations, 'key') },
+        projects: { ...state.projects, ...keyBy(response.projects, 'key') },
+        results,
+        selected: list.length > 0 ? list[0].key : null
+      }));
+    });
+  };
+
+  searchMore = (qualifier: string) => {
+    this.setState({ loading: true, loadingMore: qualifier });
+    const recentlyBrowsed = RecentHistory.get().map(component => component.key);
+    getSuggestions(this.state.query, recentlyBrowsed, qualifier).then(response => {
+      const group = response.results.find(group => group.q === qualifier);
+      const moreResults = (group ? group.items : []).map(item => ({ ...item, qualifier }));
+      this.setState(state => ({
+        loading: false,
+        loadingMore: null,
+        more: { ...state.more, [qualifier]: 0 },
+        organizations: { ...state.organizations, ...keyBy(response.organizations, 'key') },
+        projects: { ...state.projects, ...keyBy(response.projects, 'key') },
+        results: {
+          ...state.results,
+          [qualifier]: uniqBy([...state.results[qualifier], ...moreResults], 'key')
+        }
+      }));
+    });
+  };
+
+  handleQueryChange = (event: { currentTarget: HTMLInputElement }) => {
+    const query = event.currentTarget.value;
+    this.setState({ query });
+    if (query.length === 0) {
+      this.fetchFavorites();
+    } else if (query.length >= 2) {
+      this.search(query);
+    }
+  };
+
+  selectPrevious = () => {
+    this.setState((state: State) => {
+      const list = this.getPlainComponentsList(state.results);
+      const index = list.findIndex(component => component.key === state.selected);
+      return index > 0 ? { selected: list[index - 1].key } : undefined;
```

Comment:
I don't know, maybe not since the list can be quite big... But I was wondering since there is this behavior on some other lists...